qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/QuteLanguageServer.java

```diff
@@ -288,4 +295,16 @@ public void setTrace(SetTraceParams params) {
 		// to avoid having error in vscode, the method is implemented
 		// FIXME : implement the behavior of this method.
 	}
+
+	@Override
+	public Template getTemplate(String uri) {
+		TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+		AbstractTextDocumentService abstractTextDocumentService = textDocumentService.getTextDocumentService(textDocument);

```

Comment:
Replace with `AbstractTextDocumentService abstractTextDocumentService = textDocumentService.getTextDocumentService(uri);`

And define getTextDocumentService methods like this:

```java
private AbstractTextDocumentService getTextDocumentService(TextDocumentIdentifier document) {
		return getTextDocumentService(document.getUri());
	}

	private TextDocumentService getTextDocumentService(TextDocumentItem document) {
		return getTextDocumentService(document.getUri());
	}

	private AbstractTextDocumentService getTextDocumentService(String uri) {
		String fileExtension = getFileExtension(uri);
		if ("java".equals(fileExtension) || "class".equals(fileExtension)) {
			return javaFileTextDocumentService;
		}
		return templateFileTextDocumentService;
	}

	private static String getFileExtension(String uri) {
		if (uri == null) {
			return null;
		}
		int index = uri.lastIndexOf('.');
		return index != -1 ? uri.substring(index + 1, uri.length()) : null;
	}
```